### ISSUE TITLE
[Feature Request] Update Default Gearscript Definitions

### ISSUE THREAD

**Issue by dgibso29** [Bubbus/F3_CA_BUB#227]
**Is your feature request related to a problem? Please describe.**
A major feature of the framework is that you can run a mission "out of the box" if desired. This is great for new mission makers and for putting on missions with minimal effort. 

Unfortunately, I've recently relied on these gearscripts and found a mixture of issues, mostly around medical equipment in odd or missing quantities. The Blufor Guerilla preset is especially outdated.

**Describe the solution you'd like**
All four default Gearscript definitions need to be given a detailed overhaul, ensuring consistent medical equipment and roles.

We should consider adding a few new default roles while we're here -- GREN, MMG, MAT, associated Assistants, DEMO (separate of engineer), and any other common roles not currently represented. We should have these roles present in default groups as well, perhaps as a weapons team or two. Providing these roles will help newer mission makers provide a wider variety of roles with greater confidence, and enhances the frameworks ability to provide quality "out of the box" events.

**Why is this this best way to solve the problem?**
Default framework loadouts set a standard for the community to follow. We need to ensure they're up-to-date and comprehensive. 

**Additional context**
This is a great opportunity to set a standard for distribution of gear such as entrenching tools, binoculars, and varying fluid types for medical roles. These items, and others, can have a significant impact on player experience in ways that are easily overlooked, especially entrenching tools.

The team should discuss this to identify issues we've seen in ops that could be resolved in the default loadouts.

**Comment by dgibso29** [comment]
**WIP COMMENT -- Will Create as Wiki Page once complete & reviewed by team**

# Gearscript Guidelines
What follows is the recommended baseline for creating a gearscript for your missions, or the community gearscript library. These guidelines will ensure the basics are covered while still ensuring appropriate balance for the average mission using ACE Advanced Medical.

## Base Loadout Requirements
The following should be present on all loadouts:
- Compass
- Map
- Map Tools
- Maglite OR Weapon Light
- Canteen
- Watch
- Radios (1x 343, 1x 152)
- Medical Supplies as described below

### Medical Supplies
#### Non-Medical Roles
- 1L Saline
- 2 Painkiller packs
- 2 Tourniquets
- 1 Splint
- 5 Basic Bandages
- 1 Packing Bandage
- 1 QuikClot Bandage

#### Medical Roles
##### Surgeon
The Surgeon is the ultimate medical role, generally operating at the Platoon level or higher. They are able to perform all medical duties, including stitching & use of PAKs. Their role is coordinate medical needs with the Commander, as well as to provide additional aid in mass casualty situations. They are often called to work on patients with extreme trauma, and thus have more fluids than bandages and other equipment. They carry Blood, for flavour.

Equipment:
- Medical Bag
- Personal Aid Kit
- Surgical Kit
- 8L Blood, 1000mL
- 8L Blood, 500mL
- 20 Elastic Bandages
- 20 Packing Bandages
- 10 Splints
- 8 Tourniquets
- 10 Morphine Injectors
- 10 Epinephrine Injectors
- 5 Adenosine Injectors

##### Medic
Medics operate at the Squad level, and are able to stitch wounds in addition to providing first-aid. They generally carry extra fluids and medications. Their role is to get casualties back in the fight, and to provide longer-term care like stitching when conditions allow it. They carry Saline, for flavour, and an Entrenching Tool to allow building cover.

Equipment:
- Medical Bag
- Entrenching Tool
- Surgical Kit
- 6L Saline, 1000mL
- 6L Saline, 500mL
- 25 Elastic Bandages
- 15 Packing Bandages
- 8 Splints
- 8 Tourniquets
- 8Morphine Injectors
- 8Epinephrine Injectors
- 5 Adenosine Injectors

##### Combat Lifesaver
The CLS is embedded at the fire-team level, and is generally intended to be a "Rifleman with blood & bandages." Their role is to stabilize wounded and get them back in the fight, or evacuate them to an aid station if warranted. They carry some Saline and meds. They do not carry entrenching tools.

Equipment:
- 4L Saline, 1000mL
- 4L Saline, 500mL
- 10 Elastic Bandages
- 10 Packing Bandages
- 5 QuikClot Bandages
- 4 Splints
- 6 Tourniquets
- 4 Morphine Injectors
- 4 Epinephrine Injectors
- 2 Adenosine Injectors

## Role-Specific Recommendations
### Infantry
#### Rifleman
#### Autorifleman
#### Assistant Autorifleman
#### Rifleman, Anti-Tank
#### Rifleman, Anti-Air
#### Grenadier
#### Marksman
#### Demolitions Expert
#### Medium Machinegunner
#### Assistant Medium Machinegunner
#### Medium Anti-Tank
#### Assistant Medium Anti-Tank
#### Ammo Bearer

### Crew
#### CSW Gunner
#### CSW Assistant
#### Vehicle Commander
#### Vehicle Crew
#### Helo Pilot
#### Helo Crew
#### Fixed-Wing Pilot


### Leadership & Support
#### Fireteam Leader
#### Squad Leader
#### Commander
#### Forward Air Controller / Forward Observer
#### Engineer

### Misc
#### Sniper
#### Spotter

## Supply Crates
### Small
### Medium
### Vehicle

**Comment by Bubbus** [comment]
If we're going to expand the default gearscript roles we should make it very clear in the file that whatever isn't required should be deleted.  New gearscripts are a copy/paste process and I don't want MMs to feel like they have to create a CSW gunner in order to create a successful gearscript.

On another note though, now that we have the technology to never overfill a unit loadout we could return to the F3_CA roots a little by making medical loadout separately defined and declarative.

```
CREATE_LOADOUT(med,_loadout);
ADD_MEDIC_MEDICAL_ITEMS(med);

CREATE_LOADOUT(sgn,_loadout);
ADD_SURGEON_MEDICAL_ITEMS(med);

CREATE_LOADOUT(rif,_loadout);
ADD_STANDARD_MEDICAL_ITEMS(rif);
```

This obviously doesn't help save time in the short term because it'll be a new feature and it'll need documentation.  But we do have all the building blocks in place now to support this easily.

**Comment by dgibso29** [comment]
> I don't want to confuse MMs

Yep, with you. What I'm drafting above is meant to be a wider reference. In the loadouts themselves I'd add the extra roles & put them at the bottom marked as "Optional Roles" or w/e. I can take the same approach in the documentation I'm writing as well, though I think just having a very clear "Only make the roles you need for your mission" is sufficient.

Medical idea is good, could be a neat pairing with the work I'm doing if it's something you're able to do anytime soon. Otherwise, good for later!

**Comment by dgibso29** [comment]
Posted what I already have [in the wiki](https://github.com/CombinedArmsGaming/CAFE3/wiki/Gearscript-Guidelines) since there are a lot of new folks who could benefit.
